Source incompatibilities between Mill 0.11.0-M8 and -M9

`import mill.define.Input` can no longer be resolved. This results in some hassles when cross-compiling against multiple Mill versions.

```
 [info] compiling 4 Scala sources to /home/runner/work/mill-vcs-version/mill-vcs-version/out/core/0.11.0-M8-19-514e01/compile.dest/classes ...
[warn] -target is deprecated: Use -release instead to compile against the correct platform API.
Error:  /home/runner/work/mill-vcs-version/mill-vcs-version/core/src/de/tobiasroeser/mill/vcs/version/VcsVersion.scala:5:8: object Input is not a member of package mill.define
Error:  import mill.define.{Discover, ExternalModule, Input, Module}
Error:         ^
Error:  /home/runner/work/mill-vcs-version/mill-vcs-version/core/src/de/tobiasroeser/mill/vcs/version/VcsVersion.scala:19:17: not found: type Input
Error:    def vcsState: Input[VcsState] = T.input { calcVcsState(T.log) }
Error:                  ^
[warn] one warning found
Error:  two errors found
1 targets failed
core[0.11.0-M8-19-514e01].compile Compilation failed
Error: Process completed with exit code 1.
```

* See various commit of https://github.com/lefou/mill-vcs-version/pull/109

Solution:

Keep some deprecated forwarders in `mill.define` package.

--- a/main/define/src/mill/define/package.scala
+++ b/main/define/src/mill/define/package.scala
@@ -1,0 +1,13 @@
+package mill
+
+package object define {
+  @deprecated("Use mill.define.Target instead.", "Mill after 0.11.0-M8")
+  type Input[T] = Target[T]
+  @deprecated("Use mill.define.Target instead.", "Mill after 0.11.0-M8")
+  type Persistent[T] = Target[T]
+  @deprecated("Use mill.define.Target instead.", "Mill after 0.11.0-M8")
+  type Source = Target[mill.api.PathRef]
+  @deprecated("Use mill.define.Target instead.", "Mill after 0.11.0-M8")
+  type Sources = Target[Seq[mill.api.PathRef]]
+
+}

